Assets/Scripts/Contract/ContractController.cs

```diff
@@ -60,10 +62,34 @@ public void LogAllContracts()
         foreach(KeyValuePair<string, Contract> entry in contracts)
         {
             Debug.Log("Contract Name: " + entry.Value.GetName());
-            Debug.Log("Contract Time: " + entry.Value.GetTime());
+            Debug.Log("Contract Time to Complete: " + entry.Value.GetTimeToComplete());
             Debug.Log("Assigned Workers: " + entry.Value.GetAssignedWorkers());
             Debug.Log("Award: " + entry.Value.GetAward());
             Debug.Log("Contract Type: " + entry.Value.GetContractType());
         }
     }
+
+    //progress contract if workers are assigned
+    public void ContractProgression(string guid)
+    {
+        Contract c = contracts[guid];
+
+        if (c.GetAssignedWorkers() != null)
+        {
+            c.SetElapsedTime();
+            Debug.Log("progressing" + c.GetElapsedTime());
+        }
+    }
+
+    //check for contract completion
+    public void ContractCompletion(string guid)
+    {
+        Contract c = contracts[guid];
+
+        if (c.GetTimeToComplete() <= c.GetElapsedTime())
+        {
+            c.RemoveWorkers();
+            Debug.Log("Complete");
+        }
```

Comment:
Awesome!

I have a feeling that time math might not work well with some of the balancing ideas I had. (e.g. 2x worker output for good fit employees) If it's not too hard it might be a good idea to switch to a "total effort" count instead. 

So instead of adding on the time delta, you'd subtract the effort the worker puts in from the total.

Looks good for now, but we might want to rewire this early on if balancing gets wonky.